splice_machine/src/main/java/com/splicemachine/pipeline/foreignkey/actions/OnDeleteAbstractAction.java

```diff
@@ -0,0 +1,229 @@
+            readUncommittedFilter = new SimpleTxnFilter(Long.toString(indexConglomerateId), ((WritableTxn) txnView).getReadUncommittedActiveTxn(), NoOpReadResolver.INSTANCE, SIDriver.driver().getTxnStore());
+        } else {
+            throw new IOException("invalidTxn,");
+        }
+        return Pair.newPair(readCommittedFilter, readUncommittedFilter);
+    }
+
+    private byte[] isVisible(List<DataCell> next, SimpleTxnFilter txnFilter) throws IOException {
+        int cellCount = next.size();
+        for(DataCell dc:next){
+            DataFilter.ReturnCode rC = txnFilter.filterCell(dc);
+            switch(rC){
+                case NEXT_ROW:
+                    return null; //the entire row is filtered
+                case SKIP:
+                case NEXT_COL:
+                case SEEK:
+                    cellCount--; //the cell is filtered
+                    break;
+                case INCLUDE:
+                case INCLUDE_AND_NEXT_COL: //the cell is included
+                default:
+                    break;
+            }
+        }
+        if(cellCount > 0) {
+            return next.get(0).key();
+        }
+        return null;
+    }
+
+    private Partition getTable() throws IOException {
+        if(indexTable == null) {
+            indexTable = SIDriver.driver().getTableFactory().getTable(Long.toString((backingIndexConglomId)));
+        }
+        return indexTable;
+    }
+
+    protected abstract WriteResult handleExistingRow(byte[] indexRow, byte[] sourceRowKey) throws Exception;
+
+    protected static byte[] toChildBaseRowId(byte[] indexRowId, DDLMessage.FKConstraintInfo fkConstraintInfo) throws StandardException {
+        MultiFieldDecoder multiFieldDecoder = MultiFieldDecoder.create();
+        TypeProvider typeProvider = VersionedSerializers.typesForVersion(fkConstraintInfo.getParentTableVersion());
+        int position = 0;
+        multiFieldDecoder.set(indexRowId);
+        for (int i = 0; i < fkConstraintInfo.getFormatIdsCount(); i++) {
+            if (multiFieldDecoder.nextIsNull()) {
+                throw StandardException.newException(String.format("unexpected index rowid format %s", Bytes.toHex(indexRowId)));
+            }
+            if (fkConstraintInfo.getFormatIds(i) == StoredFormatIds.SQL_DOUBLE_ID) {
+                position += multiFieldDecoder.skipDouble();
+            } else if (fkConstraintInfo.getFormatIds(i) == StoredFormatIds.SQL_REAL_ID) {
+                position += multiFieldDecoder.skipFloat();
+            } else if (typeProvider.isScalar(fkConstraintInfo.getFormatIds(i))) {
+                position += multiFieldDecoder.skipLong();
+            } else {
+                position += multiFieldDecoder.skip();
+            }
+        }
+        int lastKeyIndex = position - 2;
+
```

Comment:
Can you comment the magic number 2?

Reply:
I simplified it a bit and added more documentation.